### PR TITLE
Web response object.common merge partials

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Merge partial classes  in WebResponseObject.Common and reorder properties

Justification: `// TODO: Merge Partials `

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 1]
```diff
@@ -64,79 +51,46 @@ public long RawContentLength
...
-        private void InitializeContent()
-        {
-            this.Content = this.RawContentStream.ToArray();
-        }
-
-        private static bool IsPrintable(char c)
+        public long RawContentLength
         {
-            return (char.IsLetterOrDigit(c) || char.IsPunctuation(c) || char.IsSeparator(c) || char.IsSymbol(c) || char.IsWhiteSpace(c));
+            get { return (RawContentStream == null ? -1 : RawContentStream.Length); }
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:59]: Here and below we could remove extra braces and simplify code:
```suggestion
            get => RawContentStream is null ? -1 : RawContentStream.Length;
```

**Comment by xtqqczze** [comment]
@CarloToso You should add a justification to the description:

https://github.com/PowerShell/PowerShell/blob/3dc95ced87d38c347a9fc3a222eb4c52eaad4615/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs#L105

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 2]
```diff
@@ -64,79 +51,46 @@ public long RawContentLength
...
+            get => RawContentStream is null ? -1 : RawContentStream.Length;
         }
 
         /// <summary>
-        /// Returns the string representation of this web response.
+        /// Gets the response body content as a <see cref="MemoryStream"/>.
         /// </summary>
-        /// <returns>The string representation of this web response.</returns>
-        public sealed override string ToString()
+        public MemoryStream RawContentStream
```
Comment (xtqqczze) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:65]: `RawContentStream` (and maybe others) can be an auto-implemented property (removing the field).

**Comment by xtqqczze** [comment]
We should not remove properties if they are a public API, i.e. https://learn.microsoft.com/dotnet/api/microsoft.powershell.commands.webresponseobject.statuscode

**Comment by xtqqczze** [comment]
This PR is hard to review. I suggest splitting the changes into a minimal PR merging the partials and a follow-up PR.

**Comment by CarloToso** [comment]
> We should not remove properties if they are a public API, i.e. https://learn.microsoft.com/dotnet/api/microsoft.powershell.commands.webresponseobject.statuscode
> 
> > **[WebResponseObject.StatusCode Property (Microsoft.PowerShell.Commands)](https://learn.microsoft.com/en-us/dotnet/api/microsoft.powershell.commands.webresponseobject.statuscode?view=powershellsdk-7.3.0)**Gets the response status code.

You are right, I rollbacked the change almost immediately.

**Comment by CarloToso** [comment]
> This PR is hard to review. I suggest splitting the changes into a minimal PR merging the partials and a follow-up PR.

I know it looks hard to review, but I just merged the partials, rearranged Properties, Constructors and Methods in alphabetical order and followed @iSazonov 's advice. 
In any case I can split it if needed.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs (hunk 2 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: Done for `RawContentStream`, I will do the same for `Headers`

**Comment by iSazonov** [comment]
> In any case I can split it if needed.

There are three options:
1. Make some simple PRs.
2. Make one PR with some simple/atomic commit.
3. Use both.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 3]
```diff
@@ -33,16 +33,9 @@ public class WebResponseObject
         /// </summary>
         public Dictionary<string, IEnumerable<string>> Headers
         {
-            get
-            {
-                _headers ??= WebResponseHelper.GetHeadersDictionary(BaseResponse);
-
-                return _headers;
-            }
+            get { return WebResponseHelper.GetHeadersDictionary(BaseResponse); }
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:36]: You removed a _cache_.

**Comment by CarloToso** [comment]
I'm closing this PR, I will rewrite it better with atomic commits

**Pull request closed without merging**